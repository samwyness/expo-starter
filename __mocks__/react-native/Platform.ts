export const mockPlatform = (os: string, version?: string | number) => {
  jest.doMock('react-native/Libraries/Utilities/Platform', () => ({
    ...jest.requireActual('react-native/Libraries/Utilities/Platform'),
    OS: os,
    select: (osList: Record<string, unknown>) => osList[os],
    Version: version,
  }));
};
