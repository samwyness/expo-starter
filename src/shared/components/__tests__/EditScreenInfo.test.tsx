import { render } from '@testing-library/react-native';

import EditScreenInfo from '../EditScreenInfo';

it(`renders correctly`, () => {
  // Arrange
  const tree = render(<EditScreenInfo path="some-path" />).toJSON();

  // Act

  // Assert
  expect(tree).toMatchSnapshot();
});
