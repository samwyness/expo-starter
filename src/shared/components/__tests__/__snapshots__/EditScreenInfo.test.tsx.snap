// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "marginHorizontal": 50,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "rgba(0,0,0,0.8)",
          },
          {
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      Open up the code for this screen:
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0.05)",
          },
          [
            {
              "borderRadius": 3,
              "paddingHorizontal": 4,
            },
            {
              "marginVertical": 7,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            [
              undefined,
              {
                "fontFamily": "SpaceMono",
              },
            ],
          ]
        }
      >
        some-path
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "rgba(0,0,0,0.8)",
          },
          {
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      Change any of the text, save the file, and your app will automatically update.
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "marginHorizontal": 20,
          "marginTop": 15,
        },
      ]
    }
  >
    <Text
      href="https://docs.expo.io/get-started/create-a-new-app/#opening-the-app-on-your-phonetablet"
      onPress={[Function]}
      role="link"
      style={
        {
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#2f95dc",
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Tap here if your app doesn't automatically update after making changes
      </Text>
    </Text>
  </View>
</View>
`;
